src/apps/lwaftr/lwaftr.lua

```diff
@@ -307,13 +307,30 @@ local function cannot_fragment_df_packet_error(lwstate, pkt)
                                  ethernet_header_size, icmp_config)
 end
 
--- Given a packet containing IPv4 and Ethernet, encapsulate the IPv4 portion.
-local function ipv6_encapsulate(lwstate, pkt, next_hdr_type, ipv6_src, ipv6_dst,
-                                 ether_src, ether_dst)
-   -- TODO: decrement the IPv4 ttl as this is part of forwarding
-   -- TODO: do not encapsulate if ttl was already 0; send icmp
+local function encapsulate_and_transmit(lwstate, pkt, ipv6_dst, ipv6_src)
+   -- Do not encapsulate packets that now have a ttl of zero or wrapped around
+   local ttl = decrement_ttl(pkt)
+   if ttl == 0 or ttl == 255 then
+      if lwstate.policy_icmpv4_outgoing == lwconf.policies['DROP'] then
+         return
```

Comment:
I think you are right, and what's more, we also need to drop the packet in the other case where we do send an error ICMPv4 packet.